master/internal/rm/kubernetesrm/pods.go

```diff
@@ -918,28 +919,29 @@ func (p *pods) cleanUpPodHandler(ctx *actor.Context, podHandler *actor.Ref) erro
 }
 
 func (p *pods) handleAPIRequest(ctx *actor.Context, apiCtx echo.Context) {
+	fmt.Println("in handle API request")
 	switch apiCtx.Request().Method {
 	case echo.GET:
-		summary, err := p.summarize(ctx)
-		if err != nil {
-			ctx.Respond(apiCtx.JSON(http.StatusInternalServerError, err))
+		summaries := p.summarizeClusterByNodes(ctx)
+		_, nodesToPools := p.getNodeResourcePoolMapping(summaries)
+		for nodeName, summary := range summaries {
+			summary.ResourcePool = "kuber" + strings.Join(nodesToPools[summary.ID], ",")
+			summaries[nodeName] = summary
 		}
-		ctx.Respond(apiCtx.JSON(http.StatusOK, summary))
+		ctx.Respond(apiCtx.JSON(http.StatusOK, summaries))
 	default:
 		ctx.Respond(echo.ErrMethodNotAllowed)
 	}
 }
 
 func (p *pods) handleGetAgentsRequest(ctx *actor.Context) {
-	summaries, err := p.summarize(ctx)
-	if err != nil {
-		ctx.Respond(err)
-		return
-	}
+	fmt.Println("in handle Get agents")
+	nodeSummaries := p.summarizeClusterByNodes(ctx)
+	_, nodesToPools := p.getNodeResourcePoolMapping(nodeSummaries)
 
 	response := &apiv1.GetAgentsResponse{}
-
-	for _, summary := range summaries {
+	for _, summary := range nodeSummaries {
+		summary.ResourcePool = "kuber" + strings.Join(nodesToPools[summary.ID], ",")
```

Comment:
what's with the 'kuber'? also, i think we should probably just make `AgentSummary.ResourcePools` a `[]string` for now.